huaweicloud/services/ecs/data_source_huaweicloud_compute_instance_remote_console.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API ECS POST /v1/{project_id}/cloudservers/{server_id}/remote_console
+func DataSourceComputeInstanceRemoteConsole() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceComputeInstanceRemoteConsoleRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			// attributes
+			"remote_console": computedSchemaRemoteConsole(),
+		},
+	}
+}
+
+func computedSchemaRemoteConsole() *schema.Schema {
+	computedSchema := schema.Schema{
+		Type:     schema.TypeList,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"id": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"protocol": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"type": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"url": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+
+	return &computedSchema
+}
+
+func dataSourceComputeInstanceRemoteConsoleRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	instanceId := fmt.Sprintf("%v", d.Get("instance_id"))
+	product := "ecs"
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating ECS client: %s", err)
+	}
+
+	requestOpts := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody: map[string]interface{}{
+			"remote_console": utils.RemoveNil(map[string]interface{}{
+				"protocol": "vnc",
+				"type":     "novnc",
+			}),
+		},
+	}
+
+	getVNCRemoteAddressHttpUrl := "v1/{project_id}/cloudservers/{server_id}/remote_console"
+	getVNCRemoteAddressPath := client.Endpoint + getVNCRemoteAddressHttpUrl
+	getVNCRemoteAddressPath = strings.ReplaceAll(getVNCRemoteAddressPath, "{project_id}", client.ProjectID)
+	getVNCRemoteAddressPath = strings.ReplaceAll(getVNCRemoteAddressPath, "{server_id}", instanceId)
+	getVNCRemoteAddressResp, err := client.Request("POST", getVNCRemoteAddressPath, &requestOpts)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving VNC Remote Address")
+	}
+	getVNCRemoteAddressBody, err := utils.FlattenResponse(getVNCRemoteAddressResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(instanceId)
+
+	var mErr *multierror.Error
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("remote_console", flattenListDnatRuleResponseBody(getVNCRemoteAddressBody)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenListDnatRuleResponseBody(resp interface{}) []interface{} {
```

Comment:
it seems that the function name should be updated.

Reply:
done